WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/AnalogWatchCanvasRenderer.kt

```diff
@@ -0,0 +1,546 @@
+    // associated [AnalogWatchFaceEntity] in the database.
+    private suspend fun updateColorStyleAndHourPips(
+        userStyle: UserStyle,
+        analogWatchFaceEntity: AnalogWatchFaceEntity
+    ) {
+        var revisedAnalogWatchFaceEntity = analogWatchFaceEntity
+
+        // 1. Updates color style associated with the watch face.
+        val newColorStyleSetting: String =
+            userStyle.toMap()[COLOR_STYLE_SETTING] ?: revisedAnalogWatchFaceEntity.activeColorStyleId
+
+        if (revisedAnalogWatchFaceEntity.activeColorStyleId != newColorStyleSetting) {
+            revisedAnalogWatchFaceEntity =
+                revisedAnalogWatchFaceEntity.copy(activeColorStyleId = newColorStyleSetting)
+
+            val complicationStyleDrawableId =
+                analogWatchFaceViewModel.getWatchFaceColorStyles(newColorStyleSetting).complicationStyleDrawableId
+
+            // Apply the color style to the complications. ComplicationDrawables for each of
+            // the styles are defined in XML so we need to replace the complication's
+            // drawables.
+            for ((_, complication) in complicationsManager.complications) {
+                complication.renderer = CanvasComplicationDrawable(
+                    ComplicationDrawable.getDrawable(
+                        context,
+                        complicationStyleDrawableId)!!,
+                    watchState
+                )
+            }
+        }
+
+        // 2. Update Hour Pips (dashes around the outside of the watch).
+        val newHoursPipsSettingStringVersion: String =
+            userStyle.toMap()[DRAW_HOUR_PIPS_STYLE_SETTING] ?: ""
+
+        val newHoursPipsSetting =
+            if (newHoursPipsSettingStringVersion.isEmpty()) {
+                revisedAnalogWatchFaceEntity.drawHourPips
+            } else {
+                newHoursPipsSettingStringVersion.toBoolean()
+            }
+
+        if (revisedAnalogWatchFaceEntity.drawHourPips != newHoursPipsSetting) {
+            revisedAnalogWatchFaceEntity =
+                revisedAnalogWatchFaceEntity.copy(drawHourPips = newHoursPipsSetting)
+        }
+
+        analogWatchFaceViewModel.updateAnalogWatchFace(revisedAnalogWatchFaceEntity)
+    }
+
+    // Updates the length of the minute hand arm dimensions in the associated
+    // [WatchFaceArmDimensionsEntity] in the database.
+    private suspend fun updateMinuteArmLength(
+        userStyle: UserStyle,
+        minuteHandDimensionsId: String
+    ) {
+        val newHandLengthSettingStringVersion: String =
+            userStyle.toMap()[WATCH_HAND_LENGTH_STYLE_SETTING] ?: ""
+
+        if (newHandLengthSettingStringVersion.isNotEmpty()) {
+            val newMinuteHandLength: Float =
+                newHandLengthSettingStringVersion.toFloat()
+
+            Log.d(TAG, "newMinuteHandLength: $newMinuteHandLength")
+
+            val minuteWatchFaceArmDimensions: WatchFaceArmDimensionsEntity =
```

Comment:
nit: Likewise WatchFaceArmDimensionsEntity is not required but it might be a useful as a 'comment'.

Reply:
I did put some of these in exactly for that, since I thought it would help map them to the custom class.